Enhancements for running MinIO in different modes

daveram opened the pull request:
Includes an extra var, MINIO_MODE to determine whether MinIO runs in server or gateway mode with a specific cloud provider gateway argument, eg "gateway azure".  Also included other minor fixes such as metadata and label cleanup so templates can also be deployed through the OpenShift web console, and default parameter values that align with MinIO

Commit:
Enhancements for running MinIO in different modes

Includes an extra var, MINIO_MODE to determine whether MinIO runs in server or gateway mode with a specific cloud provider gateway argument, eg "gateway azure".  Also included other minor fixes such as metadata and label cleanup so templates can also be deployed through the OpenShift web console, and default parameter values that align with MinIO

daveram commented:
As the default environment variable in the DockerFile is set to "server" this ensures it doesn't break any existing deployments that won't yet know to set MINIO_MODE when deploying.  This allows more advanced MinIO users who want to start using it as a gateway the option to do so without having to break the entrypoint on purpose in their deployment.  For an example, see https://github.com/bcgov/cap-eagle-helper-pods/blob/master/openshift/templates/api/cap/minio-eagle-cwu-azure.dc.yaml

cvarjao commented:
@daveram , can you please update `.pipeline/pipeline-cli` line to to read:
```
export OCP_PIPELINE_VERSION='0.0.7'
```

It is not exactly related to your changes, but this repo hasn't been built in a while and we need to update the scripts so that it can successfully build/deploy.

Commit:
Update pipeline-cli

As this repo hasn't been built in a while, so that it can successfully build/deploy

daveram commented:
Commit added for pipeline-cli

Commit:
Removed MINIO_CONFIG_DIR as it's deprecated

First paragraph here https://docs.min.io/docs/minio-server-configuration-guide.html

Config is now stored in configured backend storage instead of on the filesystem

daveram commented:
config dir parameter was causing issues with gateway mode, turns out this has been deprecated as config is now stored on the backend.  Linked to MinIO docs as part of commit, tested with a local image build from my github fork with success

daveram commented:
Also I checked to see if anyone was using the config dir for anything they wanted to keep, looks like we're in the clear
https://github.com/search?q=org%3Abcgov+MINIO_CONFIG_DIR&type=Code
